From 6f13b0f7e748c51b456f32c134dc3bb2873dce6b Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:34 -0400
Subject: [usb] usbtest: add pattern check on pipe in phase of unlink read

Message-id: <1410887463-20674-503-git-send-email-dzickus@redhat.com>
Patchwork-id: 95386
O-Subject: [RHEL7 PATCH 502/591] usb: usbtest: add pattern check on pipe in phase of unlink read
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit a7683eb3af6df30e3a3f3666d834e9abba8f5c22
Author: Huang Rui <ray.huang@amd.com>
Date:   Thu May 22 18:06:14 2014 +0800

    usb: usbtest: add pattern check on pipe in phase of unlink read

    TEST 11 unlinks the URB read request for N times. When host and gadget
    both initialize pattern 1 (mod 63) data series to do IN transfer, the
    host side function should check the data buffer if it is as mod 63
    series, because the data packet which host receivced will follow
    pattern 1. So this patch adds this checking action.

    Signed-off-by: Huang Rui <ray.huang@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/misc/usbtest.c b/drivers/usb/misc/usbtest.c
index 4ed457e..51a6da2 100644
--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -1345,6 +1345,9 @@ static int unlink1(struct usbtest_dev *dev, int pipe, int size, int async)
   while (!completion_done(&completion)) {
    retval = usb_unlink_urb(urb);
 
+   if (retval == 0 && usb_pipein(urb->pipe))
+    retval = simple_check_buf(dev, urb);
+
    switch (retval) {
    case -EBUSY:
    case -EIDRM:
-- 
1.7.1